endgame_simulations/common.py

```diff
@@ -0,0 +1,69 @@
+from abc import ABC, abstractclassmethod, abstractmethod
+from typing import Generic, Protocol, TypeVar
+
+import h5py
+import tqdm
+from hdf5_dataclass import FileType
+
+from endgame_simulations.models import (
+    BaseInitialParams,
+    BaseProgramParams,
+    EndgameModel,
+    InitialParams,
+    _BaseUpdateParams,
+)
+
+StateParams = TypeVar("StateParams", bound=BaseInitialParams)
+
+
+class BaseState(Generic[StateParams], ABC):
+    current_time: float
+
+    @abstractclassmethod
```

Comment:
abstractclassmethod is deprecated since 3.3. Instead:
```py
@classmethod
@abstractmethod
```
https://docs.python.org/3/library/abc.html#abc.abstractclassmethod

Reply:
Hmm ok - I guess that makes sense!